src/main/java/duke/CommandHandler.java

```diff
@@ -0,0 +1,276 @@
+        return command.substring(indexOfFirstBlankspace + 1);
+    }
+
+    private static void createNewToDo(ToDo toDo) {
+        tasks.add(toDo);
+        taskCounter++;
+        printAddedTask(toDo);
+    }
+    private static void createNewDeadline(Deadline deadline) {
+        tasks.add(deadline);
+        taskCounter++;
+        printAddedTask(deadline);
+    }
+    private static void createNewEvent(Event event) {
+        tasks.add(event);
+        taskCounter++;
+        printAddedTask(event);
+    }
+
+    private static void printAddedTask(Task task) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  [" + task.getType() + "]" + "[ ] " + task);
+        System.out.println("Now you have " + taskCounter + " tasks in your list.");
+        drawLine();
+    }
+
+    public static void executeUserCommands(DukeFile df) {
+        loadTasks(df);
+        Scanner in = new Scanner(System.in);
+        String command;
+        do {
+            command = in.nextLine();
+            drawLine();
+            if (command.equals("bye")) {
+                byeUser();
+                break;
+            }
+            else if (command.equals("list")) {
+                listTasks();
+            }
+            else if (command.contains("unmark")) {
+                int taskID = getTaskID(command);
+                try {
+                    unmarkTask(taskID);
+                } catch (DukeException e) {
+                    e.printErrorMessage();
+                    drawLine();
+                } catch (NullPointerException | IndexOutOfBoundsException e) {
+                    System.out.println(":( OOPS!!! This task does not exist.");
+                    drawLine();
+                }
+            }
+            else if (command.contains("mark")) {
+                int taskID = getTaskID(command);
+                try {
+                    markTask(taskID);
+                } catch (DukeException e) {
+                    e.printErrorMessage();
+                    drawLine();
+                } catch (NullPointerException | IndexOutOfBoundsException e) {
+                    System.out.println(":( OOPS!!! This task does not exist.");
+                    drawLine();
+                }
+            }
+            else if (command.contains("delete")) {
+                int taskID = getTaskID(command);
+                try {
+                    deleteTask(taskID);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println(":( OOPS!!! This task does not exist.");
+                    drawLine();
+                }
+            }
+            else {
+                createNewTask(command);
+            }
+            try {
+                df.saveFile(df.getFile(), getTaskListAsString());
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
```

Comment:
[Code Quality] The control flow for this whole section could be improved.

Tip: you could simplify the exception handling to a single `try-catch` block.